workflows/README.md

```diff
@@ -0,0 +1,310 @@
+# Workflow templates
+
+The content in this folder doesn't _necessarily_ belong in this repository, but we don't have a great place to put it that allows for good discoverability. The GitHub Actions workflows in this folder can be adapted to work with any CI, because the base structure will remain the same.
+
+Example implementation of these workflow files [can be found in `exercism/javascript`][git-javascript].
+
+## **HELP**: this looks like _a lot_ of work :sweat:
+
+The rest of the document is really to explain what's going on in the workflows. If you're in a hurry, and you really want to drop `.travis.yml`, but you can't invest right now in making the PR scripts are optimized, scroll down to the **~10 minute guide** on [**Migrating from Travis**](#migrating-from-travis).
+
+## Track CI actions
+
+The recommended actions are as follows:
+
+1. [`configlet` linting][git-configlet] in order to check `config.json`
+2. check for stubs
+3. check for documentation (`v3` requires new files; this might move to `configlet`)
+4. lint the exercises using a "maintainers" configuration
+5. test the exercises using the example/examplar files (can include build step)
+
+There can be track-specific actions, for example:
+
+1. check the [integrity][wiki-integrity] of the exercise configurations
+2. check the formatting of the exercise files
+
+And perhaps you'd want more Quality Of Life checks, such as:
+
+1. ensure CONTRIBUTING exists
+2. ensure there is a sane lockfile for dependencies
+3. ensure links inside markdown files are valid
+4. ...
+
+## Recommendations
+
+For each action think about how often it should run.
+
+- The `configlet` linting is something that's so important (because a track can break if the `config.json` breaks), that it probably should run, always, **but** only needs to run once per commit.
+- The existence or integrity of files only needs to run once per commit.
+- If a track is supposed to run under multiple runtime-versions or compiler-versions, building/testing exercises should be ran against each supported version
+- PRs _probably_ only need to run actions on files _added_ or _changed_, but since a file can influence an exercise, it's safer to run the actions for the _exercise_, if one of its files changes.
+
+It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. **Do not inline** the action inside the workflow files. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.
```

Comment:
```suggestion
It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. To achieve this **do not inline** the action inside the workflow files., but create a standalone script. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.
```